src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -113,27 +275,158 @@ private function getPortalTreeRootNodeId(): string
+    {
+        $treeNodeDataModel->setType('');
+
+        $children = $node->GetChildren(true);
+        if ($children->Length() > 0) {
+            $treeNodeDataModel->setType('folder');
+        }
+        if (true === $node->fieldHidden) {
+            $this->addIconToTreeNode($treeNodeDataModel, 'nodeHidden', 'fas fa-eye-slash');
+            $treeNodeDataModel->addLinkHtmlClass('node-hidden');
+        }
+        if ('' !== $node->sqlData['link']) {
+            $this->addIconToTreeNode($treeNodeDataModel, 'externalLink', 'fas fa-external-link-alt');
+            return;
+        }
+
+        if (true === $this->isPortalSelectMode && $node->id === $this->activeNodeId) {
+            $treeNodeDataModel->setSelected(true);
+        }
+
+        $this->setPageAttributes($treeNodeDataModel, $node);
+    }
+
+    private function addIconToTreeNode(BackendTreeNodeDataModel $treeNodeDataModel, string $type, string $fontawesomeIcon): void
+    {
+        if ('' === $treeNodeDataModel->getType()) {
+            $treeNodeDataModel->setType($type);
+        } else {
+            $treeNodeDataModel->addFurtherIcon('<i class="'. $fontawesomeIcon .' mr-2"></i>');
+        }
+    }
+
+    private function setPageAttributes(BackendTreeNodeDataModel $treeNodeDataModel, \TdbCmsTree $node): void
+    {
+        $linkedPageOfNode = $node->GetLinkedPageObject(true);
+        if (false === $linkedPageOfNode) {
+            if ('' === $treeNodeDataModel->getType()) {
+                $this->addIconToTreeNode($treeNodeDataModel, 'noPage', 'fas fa-genderless');
+                $this->disableSelectionWysiwyg($treeNodeDataModel);
+            }
+            return;
+        }
+
+        $this->addIconToTreeNode($treeNodeDataModel, 'page', 'far fa-file');
+        $treeNodeDataModel->addListAttribute('isPageId', $linkedPageOfNode->id);
+
+        if (true === $linkedPageOfNode->fieldExtranetPage) {
+            $treeNodeDataModel->addLinkHtmlClass('locked');
+            $this->addIconToTreeNode($treeNodeDataModel, 'locked', 'fas fa-user-lock');
+            if (false === $node->fieldShowExtranetPage) {
+                $this->addIconToTreeNode($treeNodeDataModel, 'extranetpageHidden', 'far fa-eye-slash');
+                $treeNodeDataModel->addLinkHtmlClass('extranetpage-hidden');
+            }
+        }
+        $this->setCheckStatus($treeNodeDataModel, $node->id);
+
+    }
+
+    protected function setCheckStatus(BackendTreeNodeDataModel $treeNodeDataModel, $nodeId): void
+    {
+        if ($this->activeNodeId === $nodeId) {
+            $treeNodeDataModel->setSelected(true);
+        } else {
+            if (false === $this->isPortalHomeNodeSelectMode) {
+                $treeNodeDataModel->setDisabled(true);
+                $treeNodeDataModel->addListHtmlClass('no-checkbox');
+            }
+        }
+    }
+
+    protected function disableSelectionWysiwyg(BackendTreeNodeDataModel $treeNodeDataModel): void
```

Comment:
The overwritten methods should have a /** {@inheritdoc} */ (also other locations).

Reply:
It's the other way. This is the parent function. In NavigationTreeSingleSelectWysiwyg.php comes a child, that overwrites this. How do I sign this?